code/game/turfs/closed/minerals.dm

```diff
@@ -71,16 +74,20 @@
 		if (!isturf(T))
 			return
 
-		if(last_act + (40 * I.toolspeed) > world.time)//prevents message spam
+		if(TIMER_COOLDOWN_CHECK(src, user)) //prevents mining turfs in progress
 			return
-		last_act = world.time
+
+		TIMER_COOLDOWN_START(src, user, tool_mine_speed)
```

Comment:
The second argument for timer cooldowns is an index

Instead of passing `user`, you should pass it a key that designates the user - `REF(user)` for example

Otherwise, you may cause some hard deletes storing user in the atom's cooldown list